extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -72,12 +73,33 @@ class MapboxCarMap {
     mapInitOptions: MapInitOptions,
   ) = apply {
     check(mapInitOptions.context is CarContext) {
-      "You must setup the MapboxCarMap MapInitOptions with a CarContext"
+      "You must set up the MapboxCarMap MapInitOptions with a CarContext"
     }
     carMapSurfaceOwner.setup(carContext, mapInitOptions)
     carContext.getCarService(AppManager::class.java).setSurfaceCallback(carMapSurfaceOwner)
   }
 
+  /**
+   * Instead of using [setup], this function allows you to create your own [SurfaceCallback] and
+   * forward the calls to the [CarMapSurfaceOwner]. This makes it possible for you to adopt new
+   * api versions and continue to use the [MapboxCarMap] as designed.
+   *
+   * This is a temporary solution, while androidx.car.app:app:1.3.0 is rolling out
+   * [SurfaceCallback.onClick]. If there is no use for this function in the future, it will be
+   * removed.
+   */
+  @MapboxExperimental
+  fun setupWithCustomCallback(
```

Comment:
Was a bit confused by naming, was expecting to find `custom callback` somewhere as an argument. Maybe rename to show that this one does _not_ set the callback instead, e.g.
`setupWithoutSetSurfaceCallback` ?

Reply:
Yeah I can see why it seems there should be a `CustomCallback` with this name 🤔 

`setupWithoutSetSurfaceCallback` is a little confusing to me too. Going to list a few ideas

- setupCarMapSurfaceOwner
- prepareSurfaceCallback

Actually, while thinking about `prepareSurfaceCallback`, there is no reason to surface `CarMapSurfaceOwner`

Reply:
ok `prepareSurfaceCallback` looks much better. pushing

it also works well because `setup` can now call `prepareSurfaceCallback`, and `prepareSurfaceCallback` is just exposed so the code duplication is gone and the test coverage is better.

It also makes it so `CarMapSurfaceOwner` does not need to be surfaced as a experimental public api

Reply:
Awesome thanks for this comment! Much better :)

Reply:
> Awesome thanks for this comment! Much better :)

Agree, looks real nice now!